Add documentation for Mattermost Helm chart (#13728)
gravitational/teleport: docs/pages/reference/helm-reference/teleport-plugin-mattermost.mdx

---
title: teleport-plugin-mattermost Chart Reference
description: Values that can be set using the teleport-plugin-mattermost Helm chart
---

> The `teleport-plugin-mattermost` Helm chart is used to configure the MatterMost Teleport plugin which allows users to receive access requests to channels or as direct messages via MatterMost.

Reviewer: ```suggestion
The `teleport-plugin-mattermost` Helm chart is used to configure the Mattermost Teleport plugin, which allows users to receive Access Requests via channels or direct messages in MatterMost.
```

You can [browse the source on GitHub](https://github.com/gravitational/teleport-plugins/tree/master/charts/access/mattermost).

This reference details available values for the `teleport-plugin-mattermost` chart.

(!docs/pages/includes/backup-warning.mdx!)

## `teleport.address`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

> This parameter contains the host/port combination of the teleport auth server.

Reviewer: ```suggestion
This parameter contains the host/port combination of the Teleport Auth Service or Proxy Service.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  teleport:
    address: "teleport.example.com:3025"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set teleport.address="teleport.example.com:3025"
  ```
  </TabItem>
</Tabs>

## `teleport.identitySecretName`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

> Name of the Kubernetes secret that contains the credentials for the connection.

Reviewer: ```suggestion
Name of the Kubernetes secret that contains the credentials for the connection to your Teleport cluster.
```

The secret should be in the following format:

```yaml
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: teleport-plugin-mattermost-identity
data:
  auth_id: ...
```

> Check out the [MatterMost Helm Chart documentation](https://github.com/gravitational/teleport-plugins/tree/master/charts/access/mattermost#prerequisites) for more information about how to acquire these credentials!

Reviewer: I think we should link to the Mattermost guide here so users stay within the docs site.

> Check out the [MatterMost Helm Chart documentation](https://github.com/gravitational/teleport-plugins/tree/master/charts/access/mattermost#prerequisites) for more information about how to acquire these credentials!

Reviewer: ```suggestion
Check out the [Mattermost Helm Chart documentation](https://github.com/gravitational/teleport-plugins/tree/master/charts/access/mattermost#prerequisites) for more information about how to acquire these credentials.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  teleport:
    identitySecretName: "teleport-plugin-mattermost-identity"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set teleport.identitySecretName="teleport-plugin-mattermost-identity"
  ```
  </TabItem>
</Tabs>

## `teleport.identitySecretPath`

| Type | Default value | Required? |
| - | - | - |
| `string` | `"auth_id"` | No |

> Name of the key in the Kubernetes secret that holds the credentials for the connection. If the secret follows the format above, it can be omitted.

Reviewer: ```suggestion
Name of the key in the Kubernetes secret that holds the credentials for the connection to your Teleport cluster. If the secret has the path, `"auth_id"`, you can omit this field.
```

Reviewer: Also, I think the difference between this and `identitySecretName` can be clearer. One is the name of a secret and one is a key in the secret's `data` field?

Author: I rephrased it a bit, I think it's clearer now

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  teleport:
    identitySecretPath: "auth_id"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set teleport.identitySecretPath="auth_id"
  ```
  </TabItem>
</Tabs>

## `mattermost.url`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

> Base URL of the MatterMost instance

Reviewer: Let's `s/MatterMost/Mattermost/` throughout

> Base URL of the MatterMost instance

Reviewer: ```suggestion
Base URL of the MatterMost instance.
```
For consistency with other entries

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  mattermost:
    url: "https://mattermost.example.com/"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set mattermost.url="https://mattermost.example.com/"
  ```
  </TabItem>
</Tabs>

## `mattermost.token`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | No |

> MatterMost token of the bot user to impersonate when sending the messages.

Reviewer: ```suggestion
MatterMost token of the bot user to impersonate when sending Access Request messages.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  mattermost:
    token: "xoxb-1234"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set mattermost.token="xoxb-1234"
  ```
  </TabItem>
</Tabs>

## `mattermost.tokenFromSecret`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | No |

Secret containing the MatterMost token of the bot user.

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  mattermost:
    tokenFromSecret: "teleport-mattermost-plugin-token"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set mattermost.tokenFromSecret="teleport-mattermost-plugin-token"
  ```
  </TabItem>
</Tabs>

## `mattermost.tokenSecretPath`

| Type | Default value | Required? |
| - | - | - |
| `string` | `"mattermostToken"` | No |

Key where the token is located inside the secret specified by `mattermost.tokenFromSecret`.

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  mattermost:
    tokenSecretPath: "token"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set mattermost.tokenSecretPath="token"
  ```
  </TabItem>
</Tabs>

## `mattermost.recipients`

| Type | Default value | Required? |
| - | - | - |
| `array` | `[]` | Yes |

> List of recipients who will receive notifications about access requests.

Reviewer: ```suggestion
List of recipients who will receive notifications about Access Requests.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  mattermost:
    recipients:
      - user1@example.com
      - user2@example.com
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set mattermost.recipients[0]="user1@example.com",mattermost.recipients[0]="user1@example.com"
  ```
  </TabItem>
</Tabs>

## `log.output`

| Type | Default value | Required? |
| - | - | - |
| `string` | `stdout` | No |

> Logger output. Could be "stdout", "stderr" or a file name, eg. "/var/log/teleport/mattermost.log"

Reviewer: I think we should format the possible values in code style so it's more explicit that these are possible values.

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  log:
    output: /var/log/teleport/mattermost.log
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set log.output="/var/log/teleport/mattermost.log"
  ```
  </TabItem>
</Tabs>

## `log.severity`

| Type | Default value | Required? |
| - | - | - |
| `string` | `INFO` | No |

> Logger severity. Possible values are "INFO", "ERROR", "DEBUG" or "WARN".

Reviewer: As above, I think we should format the possible values in code style so it's more explicit that these are possible values.

---
title: teleport-plugin-mattermost Chart Reference
description: Values that can be set using the teleport-plugin-mattermost Helm chart
---

The `teleport-plugin-mattermost` Helm chart is used to configure the

> Mattermost Teleport plugin which allows users to receive Access

Reviewer: ```suggestion
Mattermost Teleport plugin, which allows users to receive Access
```

Requests via channels or as direct messages in Mattermost.

You can [browse the source on GitHub](https://github.com/gravitational/teleport-plugins/tree/master/charts/access/mattermost).

This reference details available values for the `teleport-plugin-mattermost` chart.

(!docs/pages/includes/backup-warning.mdx!)

## `teleport.address`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

This parameter contains the host/port combination of the Teleport Auth Service or Proxy Service.

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  teleport:
    address: "teleport.example.com:3025"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set teleport.address="teleport.example.com:3025"
  ```
  </TabItem>
</Tabs>

## `teleport.identitySecretName`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

Name of the Kubernetes secret that contains the credentials for the connection to your Teleport cluster.

The secret should be in the following format:

```yaml
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: teleport-plugin-mattermost-identity
data:
  auth_id: ...
```

> Check out [Teleport Mattermost Plugin Setup](../../access-controls/access-request-plugins/ssh-approval-mattermost/)

Reviewer: ```suggestion
Check out the [Access Requests with Mattermost](../../access-controls/access-request-plugins/ssh-approval-mattermost/) guide
```
To use the title of the guide

> for more information about how to acquire these credentials!

Reviewer: ```suggestion
for more information about how to acquire these credentials.
```
